middleware/middleware_access.go

```diff
@@ -0,0 +1,195 @@
+package middleware
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/0xsequence/quotacontrol/proto"
+
+	"github.com/go-chi/jwtauth/v5"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+// SetCredentials gets the credentials from the JWT or the Access key.
+// It uses `jwtauth.Verifier` to get the JWT, extra  the project claim and sets it in the context as well.
```

Comment:
"SetCredentials gets the credentials"

set vs. get .. can we improve the godoc please?